src/core/EdFi.DataManagementService.Core/Middleware/DisallowDuplicateReferencesMiddleware.cs

```diff
@@ -130,4 +158,30 @@ private static string ExtractArrayName(string path)
         string[] parts = path.Split('.');
         return parts.Length > 1 ? parts[1].Trim('[', ']', '*') : string.Empty;
     }
+
+    private static Dictionary<string, Dictionary<int, List<DescriptorReference>>> GroupByArrayNameAndIndex(
```

Comment:
This method is too complex to not be documented.
Example: Explain the return type. What do the `string` and `int` Dictionary keys in this collection of collections of collections represent?